docs/articles/documentation/extending-testcafe/custom-browser-provider-plugin/browser-provider-methods.md

```diff
@@ -0,0 +1,127 @@
+
+```text
+async init()
+```
+
+For example, the existing [PhantomJS browser provider](https://github.com/DevExpress/testcafe-browser-provider-phantomjs) uses this method to create a PhantomJS instance.
+
+### dispose
+
+Performs cleanup actions for the provider data. The method is called before the `testcafe` process is finished.
+
+```text
+async dispose()
+```
+
+For example, the existing [PhantomJS browser provider](https://github.com/DevExpress/testcafe-browser-provider-phantomjs) uses this method to destroy a PhantomJS instance.
+
+### getBrowserList
+
+Returns the array of names used to identify the browsers. This method is used if the provider supports multiple browsers.
+
+```text
+async getBrowserList()
+```
+
+See [Example](index.md#example).
+
+### isValidBrowserName
+
+Checks if the specified browser name exists. This method is used if the provider supports multiple browsers.
+
+```text
+async isValidBrowserName(browserName)
+```
+
+Parameter     | Type   | Description
+------------- | ------ | -------------------------------
+`browserName` | String | The browser name to validate.
+
+Returns `true` if the specified name exists, or `false` otherwise.
+
+See [Example](index.md#example).
+
+### resizeWindow
+
+Resizes the browser window to the specified width and height.
+
+```text
+async resizeWindow(id, width, height, currentWidth, currentHeight)
+```
+
+Parameter   | Type    | Description
+----------- | ------- | ----------------------------------------------------------------------------------------------------------------
+`id`        | String  | A unique identifier generated by TestCafe that you can use to identify a browser window through the life cycle.
+`width`     | Number  | The width to resize the browser window to, in pixels.
+`height`    | Number  | The height to resize the browser window to, in pixels.
+`currentWidth`   | Number | The actual width of the browser window, in pixels. This parameter is used to resize the window correctly.
+`currentHeight`  | Number | The actual height of the browser window, in pixels. This parameter is used to resize the window correctly.
+
+See [Example](index.md#example).
+
+### takeScreenshot
+
+Takes a screenshot of the browser window.
+
+```text
+async takeScreenshot(id, screenshotPath, pageWidth, pageHeight)
+```
+
+Parameter        | Type   | Description
+---------------- | ------ | -------------------------------------------------------------------------------------------------------------
+`id`             | String | A unique identifier generated by TestCafe that you can use to identify a browser window through the life cycle.
+`screenshotPath` | String | The directory path to save the screenshots to.
+`pageWidth`   | Number | The actual width of the web page opened in the browser, in pixels.
+`pageHeight`  | Number | The actual height of the web page opened in the browser, in pixels.
+
+See [Example](index.md#example).
```

Comment:
I feel puzzled about that `life cycle` you are talking about here and there. Give more details or, most probably, be more simple and don't use `life cycle`.